caliban/util.py

```diff
@@ -671,13 +676,22 @@ def validated_file(path: str) -> str:
 class TqdmFile(object):
   """Dummy file-like that will write to tqdm"""
   file = None
+  prefix = _term_move_up() + '\r'
 
   def __init__(self, file):
     self.file = file
+    self._carriage_pending = False
+
+  def write(self, line):
+    if self._carriage_pending:
```

Comment:
tqdm is a little busted; you can't directly pass carriage returns in, so I put some logic here that appends the required prefix to the NEXT line, and swaps the carriage return with a newline.

Reply:
nice, that's a tricky fix there